### PR TITLE
Cake5

### PR DESCRIPTION
**Pull request by arusinowski**
This is initial upgrade to CakePHP5.0 and phpunit 10, when accepted please merge this PR with new branch `cake5`. Thank you

### CODE REVIEW

File: src/Model/Behavior/Strategy/TuupolaStrategy.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace Muffin\Obfuscate\Model\Behavior\Strategy;
+
+use InvalidArgumentException;
+use Tuupola\Base62;
+
+/**
+ * Class TuupolaStrategy
+ */
+class TuupolaStrategy implements StrategyInterface
```
Comment (ADmad) [inline src/Model/Behavior/Strategy/TuupolaStrategy.php]: We should probably name this new class `Base64Strategy` to be future proof. Then if needed the underlying lib can be changed without renaming/deprecating the class.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -1,90 +1,120 @@
...
+ * This function is used to find the location of CakePHP whether CakePHP
+ * has been installed as a dependency of the plugin, or the plugin is itself
+ * installed as a dependency of an application.
+ */
 
-    $root = dirname(dirname(__DIR__));
-    if (is_dir($root . '/vendor/cakephp/cakephp')) {
-        return $root;
-    }
+use Cake\Cache\Cache;
```
Comment (ADmad) [inline tests/bootstrap.php]: Please update this to use the generic bootstrapping code as done for Throttle plugin.

File: src/Model/Behavior/Strategy/Base62Strategy.php [hunk 3]
```diff
@@ -7,16 +7,16 @@
 use Tuupola\Base62;
 
 /**
- * Class TuupolaStrategy
+ * Class Base62Strategy
  */
-class TuupolaStrategy implements StrategyInterface
+class Base62Strategy implements StrategyInterface
```
Comment (ADmad) [inline src/Model/Behavior/Strategy/Base62Strategy.php:12]: Filename needs to be updated to match the classname.

Comment (arusinowski) [reply src/Model/Behavior/Strategy/Base62Strategy.php:12]: Hmm, it looks like the `Base62Strategy` class name matches the filename already

Comment (ADmad) [reply src/Model/Behavior/Strategy/Base62Strategy.php:12]: I swear the diff showed me the old filename earlier :)  Anyway all good now.

File: src/Model/Behavior/Strategy/Base62Strategy.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * Class Base62Strategy
+ */
+class Base62Strategy implements StrategyInterface
+{
+    /**
+     * Obfuscator.
+     *
+     * @var \Tuupola\Base62
+     */
+    private Base62 $base62;
```
Comment (ADmad) [inline src/Model/Behavior/Strategy/Base62Strategy.php]:
```suggestion
    protected Base62 $base62;
```

File: src/Model/Behavior/Strategy/Base62Strategy.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     *  Constructor.
+     *
+     * @param string|null $set Random alpha-numeric set.
+     */
+    public function __construct(?string $set = null)
```
Comment (ADmad) [inline src/Model/Behavior/Strategy/Base62Strategy.php]:
```suggestion
    public function __construct(?string $charset = null)
```

File: src/Model/Behavior/Strategy/Base62Strategy.php (hunk 4 above)
Comment (arusinowski) [reply src/Model/Behavior/Strategy/Base62Strategy.php]: Done

File: src/Model/Behavior/Strategy/TinyStrategy.php [hunk 6]
```diff
@@ -3,27 +3,19 @@
...
 namespace Muffin\Obfuscate\Model\Behavior\Strategy;
 
-use ZackKitzmiller\Tiny;
-
 /**
  * Class TinyStrategy
+ *
+ * @deprecated Use Muffin\Obfuscate\Model\Behavior\Strategy\Base62Strategy instead
  */
 class TinyStrategy implements StrategyInterface
```
Comment (ADmad) [inline src/Model/Behavior/Strategy/TinyStrategy.php]: I am wondering if we should just drop this `TinyStragtegy`. Existing users will have to change the underlying lib anyway, so they can change the strategy name too.

Comment (arusinowski) [reply src/Model/Behavior/Strategy/TinyStrategy.php]: You're right, we may drop the TinyStrategy completely

**Pull request merged**